src/Univ2LpOracle.sol

```diff
@@ -89,7 +89,7 @@ contract UNIV2LPOracle {
         _;
     }
 
-    address public src;             // Price source
+    address public immutable src;   // Price source
```

Comment:
I don't want to be a buzz-kill, but this almost certainly changes the storage layout which we minimally wanted to preserve for `cur` and `nxt`.

Reply:
I would suggest removing `immutable`

Reply:
```suggestion
    address public src;   // Price source
```

Reply:
Good catch, but in that case I think the better thing to do is add a dummy variable to squat in that storage slot so that we still get the gas savings of `immutable`...we care fiercely about gas in these contracts, especially after Berlin bumps up `SLOAD` costs.

Reply:
Or maybe we can shift others around, I'm not picky about that, but the gas savings matters.

Reply:
Oh I see, `src` itself is arguably part of the API:
https://github.com/makerdao/osm/blob/master/src/osm.sol#L72

Even though in other cases it's a `DSValue`, not a Uniswap pair.

In that case, my suggestions is to keep `address public src;` here, and _also_ add something like `address immutable pair;` and use the latter in `seek` for gas efficiency.

Reply:
Good point on the gas savings of a new `src`.  Perhaps we declare a `address public empty` above it.

The one thing we don't really know is how may integrations are relying on checking the storage slots of contracts like this.  We know this happens with `cur` and `nxt`, but if any other integrations are looking at the storage location of `src`, we could break them.  I honestly don't know the answer to this question, other than having integrators ask me not to change the storage layout of other contracts.

Also, this is a slightly newer type of oracle, we might be able to get away with changing it now more than a few months from now.

Reply:
It doesn't move the slots.

This was already a packed variable, so it may move the data that was packed with it, but `cur` and `nxt` are still in the proper locations, and there are some tests to verify.

Reply:
Also, there's already a public api for `src()` in both cases, so I'm not as concerned about saving the slot locations here.

Also, we may want to adjust the `val` datatypes to be larger in both `cur` and `nxt` because it's a very real possibility that this could overflow with some existing uniswap exchanges (GUSD/WBTC immediately come to mind), and any integration just dumbly looking for prices in these slots will get them wrong.

In that case, I would advocate that we specifically avoid re-using slots so that integrations must account for the differences.

Reply:
If we don't care about `hop` and `zzz` moving within their slot so long as `cur` and `nxt` stay put, then I don't really have an issue here. After all, `src` has a completely different meaning in this contract than a normal OSM (it's usually a DSValue, here it's a Uniswap pool...almost an abuse of terminology to use the same name, tbh).